sdk/identity/identity/src/credentials/managedIdentityCredential/msalMsiProvider.ts

```diff
@@ -65,11 +72,15 @@ export class MsalMsiProvider {
       _options = clientIdOrOptions ?? {};
     }
     this.resourceId = _options?.resourceId;
+    this.objectId = _options?.objectId;
 
     // For JavaScript users.
-    if (this.clientId && this.resourceId) {
+    const providedIds = [this.clientId, this.resourceId, this.objectId].filter(Boolean);
+    if (providedIds.length > 1) {
       throw new Error(
```

Comment:
Curious what would happen if nothing is passed to the constructor. Shouldn't `clientId` or one of the `options` be required? But this is existing pattern anyway...

Reply:
Good Q! At most one of these user-assigned IDs can be used. If none are provided (the default case) the credential will use the system-assigned MI. That's a valid scenario 👍 It's just that when you supply more than one user-assigned identifier that we throw rather than arbitrarily pick one